src/ngx_http_modsecurity_pre_access.c

```diff
@@ -180,12 +180,17 @@ ngx_http_modsecurity_pre_access_handler(ngx_http_request_t *r)
              * it may ask for a intervention in consequence of that.
              *
              */
-            ret = ngx_http_modsecurity_process_intervention(ctx->modsec_transaction, r);
+          
+        }
+        /** 
+         *  @dennus: I'm move transaction outside cycle, because I'm think that was uselless calls
```

Comment:
Not really. The rules can set limit for the size of the body, for example. So, whenever the size is over certain threshold modsecurity takes an action. Having that check when the body is full already, it may be  too late for the rule take an action.